ddev/src/ddev/cli/env/test.py

```diff
@@ -0,0 +1,136 @@
+def test_command(
+    ctx: click.Context,
+    *,
+    intg_name: str,
+    environment: str | None,
+    args: tuple[str, ...],
+    local_dev: bool,
+    local_base: bool,
+    agent_build: str | None,
+    extra_env_vars: tuple[str, ...],
+    junit: bool,
+    python_filter: str | None,
+    new_env: bool,
+):
+    """
+    Test environments.
+
+    If no environment is specified, `active` is selected which will test all environments
+    that are currently running. You may choose `all` to test all environments whether or not
+    they are running.
+
+    Testing active environments will not stop them after tests complete. Testing environments
+    that are not running will start and stop them automatically.
+    """
+    from ddev.cli.env.start import start
+    from ddev.cli.env.stop import stop
+    from ddev.cli.test import test
+    from ddev.e2e.config import EnvDataStorage
+    from ddev.utils.ci import running_in_ci
+    from ddev.utils.structures import EnvVars
+
+    app: Application = ctx.obj
+    storage = EnvDataStorage(app.data_dir)
+    integration = app.repo.integrations.get(intg_name)
+    active_envs = storage.get_environments(integration.name)
+
+    if environment is None:
+        environment = 'all' if (not active_envs or running_in_ci()) else 'active'
+
+    if environment == 'all':
+        import json
+        import sys
+
+        with integration.path.as_cwd():
+            env_data_output = app.platform.check_command_output(
+                [sys.executable, '-m', 'hatch', '--no-color', '--no-interactive', 'env', 'show', '--json']
+            )
+            try:
+                environments = json.loads(env_data_output)
+            except json.JSONDecodeError:
+                app.abort(f'Failed to parse environments for `{integration.name}`:\n{repr(env_data_output)}')
+
+        env_names = [
+            name
+            for name, data in environments.items()
+            if data.get('e2e-env')
+            and (not (platforms := data.get('python')) or app.platform.name in platforms)
+            and (python_filter is None or data.get('python') == python_filter)
+        ]
+    elif environment == 'active':
+        env_names = active_envs
+    else:
+        env_names = [environment]
+
+    if not env_names:
+        return
+
+    app.display_header(integration.display_name)
+
+    active = set(active_envs)
+    for env_name in env_names:
+        env_active = env_name in active
+        ctx.invoke(
+            start,
+            intg_name=intg_name,
+            environment=env_name,
+            local_dev=local_dev,
+            local_base=local_base,
+            agent_build=agent_build,
+            extra_env_vars=extra_env_vars,
+            hide_help=True,
+            ignore_state=env_active,
+        )
+
+        env_data = storage.get(integration.name, env_name)
+        metadata = env_data.read_metadata()
+        try:
+            with EnvVars(metadata.get('e2e_env_vars', {})):
```

Comment:
We could also use a constant here instead of the hardcoded `e2e_env_vars` (similar to my other suggestion for `agent_type`)

Reply:
Done!